internal/serve/httphandler/assets_handler.go

```diff
@@ -90,10 +91,15 @@ func (c AssetsHandler) CreateAsset(w http.ResponseWriter, r *http.Request) {
 			return nil, fmt.Errorf("inserting new asset: %w", insertErr)
 		}
 
-		assetToAdd := &txnbuild.CreditAsset{Code: assetCode, Issuer: assetIssuer}
-		trustlineErr := c.handleUpdateAssetTrustlineForDistributionAccount(ctx, assetToAdd, nil)
-		if trustlineErr != nil {
-			return nil, fmt.Errorf("adding trustline for the distribution account: %w", trustlineErr)
+		distributionAccount, getDistAccFromCtxErr := c.DistributionAccountResolver.DistributionAccountFromContext(ctx)
+		if getDistAccFromCtxErr != nil {
+			return nil, fmt.Errorf("resolving distribution account from context: %w", getDistAccFromCtxErr)
+		}
+
+		if distributionAccount.IsStellar() {
+			if trustlineErr := c.handleUpdateAssetTrustlineForDistributionAccount(ctx, &txnbuild.CreditAsset{Code: assetCode, Issuer: assetIssuer}, nil, distributionAccount); trustlineErr != nil {
+				return nil, fmt.Errorf("adding trustline for the distribution account: %w", trustlineErr)
+			}
```

Comment:
👋 When it's not a stellar dist account type you're still updating the database as if that asset had been inserted, making it inconsistent. That doesn't sound right to me.

The [ticket](https://stellarorg.atlassian.net/browse/SDP-1183) is about disabling asset management for Circle, but what you're doing is partially managing the asset.

The same goes for DeleteAsset.